### PR TITLE
services/horizon: Use operation body to generate "claimant created" effect

### PR DESCRIPTION
**Pull request by bartekn**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Switched to using operation body conditionally to fetch claimable balance predicated.

### Why

In https://github.com/stellar/go/commit/29c16d31daa7debe6a5bb8b1e3354a8c8dc06c22 `EffectClaimableBalanceClaimantCreated` started to be generated solely from tx meta. While this approach is fine it changes the way this effect details are ingested compared to 2.8.3. This is happening because all `rel_before` predicates present in operation body are changed to `abs_before` by Stellar-Core when the tx is included in the ledger.

### Known limitations

N/A

### CODE REVIEW

**Review by paulbellamy — APPROVED** [review]

**Comment by paulbellamy** [comment]
Test would be nice...

File: services/horizon/internal/ingest/processors/effects_processor.go [hunk 1]
```diff
@@ -959,7 +959,20 @@ func (e *effectsWrapper) addClaimableBalanceEntryCreatedEffects(source *xdr.Muxe
...
 		details,
 	)
-	for _, c := range cb.Claimants {
+	// EffectClaimableBalanceClaimantCreated can be generated by
+	// `create_claimable_balance` but also by `liquidity_pool_withdraw` which
+	// also removes LP.
+	// In case of `create_claimable_balance` we use `op.Claimants` to make
+	// effects backward compatible. The reason for this is that Stellar-Core
+	// changes all `rel_before` predicated to `abs_before` so effects generated
+	// by Horizon 2.8.3 and 2.9.0 are different.
```
Comment (2opremio) [inline services/horizon/internal/ingest/processors/effects_processor.go:968]: Could you rephrase this so that future readers can understand it better? In particular what is `abs_before` and `rel_before` and why are they different depending on where we obtain the claimants from?

I have also proposed some other changes in the comment. Among other things I don't think that the versions are important, since we are releasing with backwards compatibility (but they do add to the confusion).

```suggestion
	// EffectClaimableBalanceClaimantCreated can be generated by
	// `create_claimable_balance` operation but also by `liquidity_pool_withdraw` operation
	// causing a revocation.
	// In case of `create_claimable_balance` we use `op.Claimants` to make
	// effects backward compatible. The reason for this is that Stellar-Core
	// changes all `rel_before` predicated to `abs_before`.
```

Comment (2opremio) [reply services/horizon/internal/ingest/processors/effects_processor.go:968]: Actually, including the explanation from the PR description would be enough:

> This is happening because all rel_before predicates present in operation body are changed to abs_before by Stellar-Core when the tx is included in the ledger.

**Comment by bartekn** [comment]
OK, added a test in https://github.com/stellar/go/pull/3977/commits/f3e8e8b635ba632b92c8d9af25523fe423a9964b. I thought that behaviour is already tested but it really wasn't.

Also, rephrased the comment.

**Comment by bartekn** [comment]
Run `verify-range` on a small affected range and confirmed the fix is working.
